pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -191,7 +192,9 @@ type AWSCloudConfig struct {
 	Global struct {
 		// TODO: Is there any use for this?  We can get it from the instance metadata service
 		// Maybe if we're not running on AWS, e.g. bootstrap; for now it is not very useful
-		Zone string
+		Zone     string
+		Region   string
+		Endpoint string
```

Comment:
Not in it's current implementation. The problem is that the `aws-sdk-go` calculates the endpoint from the region. We made the mistake to not follow the exact same naming scheme. It's virtually impossible to change it now, so we are stuck with configuring the endpoints for each client... :cry: 

Be careful what you wish for... :smile: We are running a home grown (so far) non-public AWS clone. The e2e test should stay on Amazon. Even the API of our cloud is compatible there are a lot of unspecced edge cases which behave slightly differently. 

I haven't looked into running the e2e tests against our cloud. My assumption was that the time to set it up would be quite prohibitive.